atlasdb-feign/src/main/java/com/palantir/atlasdb/http/FailoverFeignTarget.java

```diff
@@ -134,11 +143,11 @@ private void failoverToNextNode(ExceptionRetryBehaviour retryBehaviour) {
 
     private void checkAndHandleFailure(RetryableException ex) {
         final long fastFailoverStartTime = startTimeOfFastFailover.get();
-        final long currentTime = System.currentTimeMillis();
+        final long currentTime = currentTimeMillis.get();
         boolean failedDueToFastFailover = fastFailoverStartTime != 0
-                && (currentTime - fastFailoverStartTime) > fastFailoverTimeoutMillis;
+                && (currentTime - fastFailoverStartTime) > fastFailoverTimeoutMillis
+                && numSwitches.get() > servers.size() - 1; //try all the available servers before failing
```

Comment:
we need to change the logic a bit more, currently it is setting numSwitches to 0 after switching to a node - which is a bit weird